schema.template.graphql

```diff
@@ -126,3 +156,30 @@ type TokenWithdrawalEvent @entity {
   amount: BigInt!
   withdrawalEvent: WithdrawalEvent!
 }
+
+type ControlTransferEvent implements AccountEvent @entity {
+  id: ID! # ct-{txHash}-${logIdx}
+  timestamp: BigInt!
+  account: Account!
+
+  type: String! # initiate | cancel | transfer
```

Comment:
@wminshew what do you think about this `type` here? Have something similar for create/update split, where the two share an event.

Nothing special about it, I think this just lets us share more code. Was kinda going back and forth on doing this versus just having separate events for everything (and a bunch of nearly identical code). Might be better just to have the separate events to keep things cleaner...

Anyway, easy to go back and update whenever.

Reply:
i think this is good but you could also make the `type` a union class or enum instead of a string?